pkg/controllerhelpers/handlers.go

```diff
@@ -89,29 +89,41 @@ func (h *Handlers[T]) EnqueueAll(depth int, untypedObj kubeinterfaces.ObjectInte
 	}
 }
 
-func (h *Handlers[QT]) EnqueueOwner(depth int, obj kubeinterfaces.ObjectInterface, operation HandlerOperationType) {
-	controllerRef := metav1.GetControllerOf(obj)
-	if controllerRef == nil {
-		return
-	}
+func (h *Handlers[QT]) EnqueueOwnerIf(precondition func(QT) bool) EnqueueFuncType {
```

Comment:
We should distinguish the function name from those that do that do the work as this one returns a function - those usually have `Func` suffix. How about `EnqueOwnerWithConditionFunc`